include/emp/web/Color.hpp

```diff
@@ -0,0 +1,459 @@
+/**
+ *  @note This file is part of Empirical, https://github.com/devosoft/Empirical
+ *  @copyright Copyright (C) Michigan State University, MIT Software license; see doc/LICENSE.md
+ *  @date 2020
+ *
+ *  @file  Color.hpp
+ *  @brief Represent RGBA color, parseable from CSS strings.
+ *
+ *  Adapted largely from https://github.com/mapbox/css-color-parser-cpp/.
+ */
+
+
+#ifndef EMP_WEB_COLOR_HPP
+#define EMP_WEB_COLOR_HPP
+
+#include <algorithm>
+#include <cstdint>
+#include <sstream>
+#include <string>
+
+#if __has_include(<SFML/Color.hpp>)
+#include <SFML/Color.hpp>
+#endif
+
+#include "../base/vector.hpp"
+#include "../geometry/Circle2D.hpp"
+#include "../tools/string_utils.hpp"
+#include "color_logic_utils.hpp"
+#include "color_map.hpp"
+
+namespace emp {
+namespace web {
+
+  /// Represents RGBA color.
+  class Color {
+    private:       
+        unsigned char r{}, g{}, b{};
+        float a{ 1.0f };
+        std::string css_str = "";
```

Comment:
maybe we should label this variable cached_css_str

and we should probably cache a sf::Color too.